src/engine/SmtCore.cpp

```diff
@@ -62,10 +62,9 @@ void SmtCore::reportViolatedConstraint( PiecewiseLinearConstraint *constraint )
          _constraintViolationThreshold )
     {
         _needToSplit = true;
-        if ( GlobalConfiguration::SPLITTING_HEURISTICS == DivideStrategy::ReLUViolation )
+        if ( GlobalConfiguration::SPLITTING_HEURISTICS ==
+             DivideStrategy::ReLUViolation || !pickSplitPLConstraint() )
             _constraintForSplitting = constraint;
-        else
```

Comment:
no else needed ?

Reply:
Here I changed it such that if the alternative heuristic fail to pick a splitting ReLU (e.g., when NetworkLevelReasoner is not available), we fail back to the native ReluViolation count based  strategy.

Reply:
added a comment